inject/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java

```diff
@@ -1686,6 +1866,51 @@ private boolean isInnerType(ClassElement genericType) {
         return beanTypeInnerClasses.contains(type);
     }
 
+    @Override
+    public void visitAnnotationMemberPropertyInjectionPoint(TypedElement annotationMemberBeanType,
+                                                            String annotationMemberProperty,
+                                                            @Nullable String requiredValue,
+                                                            @Nullable String notEqualsValue) {
+        ClassElement annotationMemberClassElement = annotationMemberBeanType.getType();
+        MethodElement memberPropertyGetter = annotationMemberClassElement.getBeanProperties()
+            .stream()
+            .filter(property -> property.getSimpleName().equals(annotationMemberProperty))
+            .findFirst()
+            .flatMap(PropertyElement::getReadMethod)
+            .orElse(null);
+
+        if (memberPropertyGetter == null) {
+            final String[] readPrefixes = annotationMetadata.getValue(AccessorsStyle.class, "readPrefixes", String[].class)
+                .orElse(new String[]{AccessorsStyle.DEFAULT_READ_PREFIX});
+
+            memberPropertyGetter = annotationMemberClassElement.getEnclosedElements(ElementQuery.ALL_METHODS)
+                .stream()
+                .filter(methodElement -> NameUtils.getPropertyNameForGetter(methodElement.getSimpleName(), readPrefixes).equals(annotationMemberProperty))
+                .filter(this::isMethodAccessible)
+                .findFirst()
+                .orElse(null);
+        }
```

Comment:
I'm not sure if this is necessary, since `getBeanProperties()` already has the logic to handle `readPrefixes`

Reply:
This part is required because it allows to reference package-private and protected properties from classes in sames packages while usage of `getBeanProperties()` allows to reference public properties only. I've also faced an issue with `@ConfigurationProperties` on groovy interfaces - `getBeanProperties()` doesn't consider abstract getters as properties because of line 935 in `GroovyClassElement` (there is no such issue with JavaClassElement):

```boolean validModifiers = node.isPublic() && !node.isStatic() && !node.isSynthetic() && !node.isAbstract();```

Reply:
I see, the latter issue with Groovy seems like a bug.

Reply:
At least, it seems like removing `!node.isAbstract()` doesn't break any tests. I can remove it in this PR. However, the question about usage of this feature with package-private and protected properties is still open